backport from upstreams
http://websvn.kde.org/branches/KDE/4.1/kdepimlibs/kcal/incidenceformatter.cpp?r1=856768&r2=856767&pathrev=856768
as requested by upstream on kde-packager and other places.

--- kdepimlibs/kcal/incidenceformatter.cpp	2008/09/03 18:28:53	856767
+++ kdepimlibs/kcal/incidenceformatter.cpp	2008/09/03 18:33:54	856768
@@ -1132,6 +1132,9 @@
     IncidenceCompareVisitor() : mExistingIncidence( 0 ) {}
     bool act( IncidenceBase *incidence, Incidence *existingIncidence )
     {
+      if (!existingIncidence) {
+        return false;
+      }
       Incidence *inc = dynamic_cast<Incidence *>( incidence );
       if ( inc && inc->revision() <= existingIncidence->revision() ) {
         return false;
